site/docs/hosting/vps.md

```diff
@@ -45,131 +137,638 @@ cd /etc/systemd/system
+
+The first method is convenient when getting to know PM2.
+However, during deployment, you should use the second method, which is what we did in our case.
+
+On our server, in the directory where the bot assembly is stored, there is a file `ecosystem.config.js` with the following contents:
+
+```js
+module.exports = {
+  apps: [{
+    name: "<app-name>",
+    script: "<start-command>",
+  }],
+};
 ```
 
-#### Don't Run on Boot
+where `<app-name>` can be any identifier, and `<start command>` must be the actual command to start the bot, for example, `node bot.js` or `npm run start`, etc.
 
-```sh
-systemctl disable <service_name>
+#### Managing the Application
+
+Below are the commands you can use to control the application.
+
+```sh:no-line-numbers
+# If the `ecosystem.config.js` file is in the current directory,
+# you can specify nothing to start the application.
+# If the application is already running, this command will restart it.
+pm2 start
+
+# All of the following commands require you to specify the name of the application
+# or the `ecosystem.config.js` file.
+# To apply the action to all applications, specify `all`.
+
+# To restart the app
+pm2 restart <<app-name>>
+
+# To reload the app
+pm2 reload <<app-name>>
+
+# To stop the app
+pm2 stop <<app-name>>
+
+# To delete an app
+pm2 delete <<app-name>>
 ```
 
-## PM2 (Node.js only)
+#### Saving the Application Operation
 
-PM2 is a daemon process manager for Node.js that will help you manage and keep your app online 24/7.
+If the server reboots, your bot will not resume working.
+For the bot to resume work, you need to prepare PM2 for this.
 
-### Installing
+On the server in the terminal, run the following command:
 
-```sh
-npm install pm2@latest -g
+```sh:no-line-numbers
+pm2 startup
+```
 
-# If using Yarn
-yarn global add pm2
+You will be provided with a command that you must execute to make PM2 automatically start after the server reboots.
+
+Then run one more command:
+
+```sh:no-line-numbers
+pm2 save
 ```
 
-### Managing Apps
+This command will save the list of current applications so that they can be launched after a server reboot.
 
-#### Start
+If you have created a new application and want to save it as well, simply run `pm2 save` again.
 
-```sh
-pm2 start --name <app_name> <entry_point>
+## Running the Bot on Webhooks
+
+To run a bot on webhooks, you will need to use a web framework and not call `bot.start()`.
+
+Here is a sample code to run the bot on webhooks that should be added to the main bot file:
+
+<CodeGroup>
+  <CodeGroupItem title="Node.js" active>
+
+```ts
+import { webhookCallback } from "grammy";
+import { fastify } from "fastify";
+
+const server = fastify();
+
+server.post(`/${bot.token}`, webhookCallback(bot, "fastify"));
+
+server.listen();
+```
+
+</CodeGroupItem>
+ <CodeGroupItem title="Deno">
+
+```ts
+import { serve } from "https://deno.land/std@0.178.0/http/server.ts";
+import { webhookCallback } from "https://deno.land/x/grammy@v1.14.1/mod.ts";
```

Comment:
These URLs are already outdated. The website will always automatically display the latest versions if we use the generic links here.
```suggestion
import { serve } from "https://deno.land/std/http/server.ts";
import { webhookCallback } from "https://deno.land/x/grammy/mod.ts";
```